File: LibraryBackEnd/library/src/main/java/com/rangan/library/model/Book.java
package com.rangan.library.model;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigInteger;

@Data
@Entity
@Table(name = "book")
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "book_name")
    private String bookName;

    @Column(name = "isbn")
    private BigInteger isbn;

    @ManyToOne
    @JoinColumn(name = "auth_id", referencedColumnName = "id")
    private Author author;
}

File: LibraryFrontEnd/app/src/main/java/com/shan/library/model/Book.kt
package com.shan.library.model

import com.squareup.moshi.Json

data class Book(
    @Json(name = "id") val id: Int,
    @Json(name = "bookName") val bookName: String,
    @Json(name = "isbn") val isbn: Long,
    @Json(name = "authorId") val authorId: Int
)